stackslib/src/chainstate/nakamoto/mod.rs

```diff
@@ -2550,6 +2550,43 @@ impl NakamotoChainState {
             &MINER_BLOCK_HEADER_HASH,
         );
 
+        if new_tenure {
+            clarity_tx
+                .connection()
+                .as_free_transaction(|clarity_tx_conn| {
+                    clarity_tx_conn.with_clarity_db(|db| {
+                        let tenure_ht = match db.get_tenure_height() {
+                            Ok(ht) => ht,
+                            Err(e) => {
+                                if cfg!(test) {
+                                    // In test mode, the epoch 3.0 initialization does not always
+                                    // get called to initialize this tenure height so we ignore
+                                    // the error here and return a default value.
+                                    warn!("Failed to get tenure height, defaulting to 0");
+                                    0
+                                } else {
+                                    // If not in test mode, this is definitely an error.
+                                    return Err(e.into());
+                                }
```

Comment:
I don't love this, but I can't find a better way to handle the test cases that do not properly call the `initialize_epoch_*` functions. Let me know if you have a better solution.

Reply:
Probably, but, for example, I was having trouble figuring out how to call it in stackslib/src/chainstate/nakamoto/coordinator/tests.rs, `advance_to_nakamoto`.

Reply:
Ah -- I think this was actually a bug.

The `if new_tenure { .. }` block should be _after_ the `process_epoch_transition` invocation.

Reply:
🤦‍♂️

Reply:
Fixed and cleaned up the tests.